### ISSUE TITLE
Using `aria-live="assertive"` for Flutter-web's `SemanticsService.announce` implementation

### ISSUE THREAD

**Issue by marcianx** [flutter/flutter#104109]
## Use case

I'm working on a custom type of editor in Flutter and need to make it accessible across various platforms (flutter-web being the focus of this issue). The editor's content is rendered as an image and renders its own custom cursor. To make it accessible, I want to vocalize similarly to how a native text field would vocalize as the user navigates through it or edits it, but adapted to the semantics of my widget.

Existing custom editors (e.g. accessible web math editors) that I have tested vocalize feedback to the end user using a hidden element with `aria-live="assertive"` on it. This vocalization happens directly due to user action and uses "assertive" to ensure that the user receives that feedback as compared to `aria-live="polite"` which doesn't seem to vocalize in some cases (especially on VoiceOver). More specifically, this editor is not a widget vocalizing background auto-updates where "poilte" would be more appropriate.

[Flutter's accessibility announcer service][announcer] seems the ideal way to do these vocalizations. However, [flutter-web implements the vocalization using an `aria-live="polite"` element][polite]. I recognize the spirit of avoiding the misuse of `"assertive"` announcements for background updates, but the case of feedback above due to direct user action, `"assertive"` seems more appropriate to ensure the user receives the feedback.

## Proposal

Please either:

1. Modify [the current implementation][polite] to use `aria-live="assertive"`, OR
2. Add an additional argument to the [announcer service][announcer] to support assertive vocalizations and implement it using an `aria-live="assertive"` element.

[announcer]: https://api.flutter.dev/flutter/semantics/SemanticsService/announce.html
[polite]: https://github.com/flutter/engine/blob/6d9f025279610ad44634cbd4f88e3f05594d009d/lib/web_ui/lib/src/engine/semantics/accessibility.dart#L74

**Comment by goderbauer** [comment]
/cc @yjbanov @dnfield @chunhtai 

I wonder, do iOS/Android/Desktop have a polite way of announcing something via the announce API? If not, we'd have to figure out what to do when polite is specified on those platforms if we go with option 2.

**Comment by chunhtai** [comment]
I have not tested it but there is this iOS https://developer.apple.com/documentation/uikit/uiaccessibilityspeechattributequeueannouncement?language=objc

**Comment by goderbauer** [comment]
I think I am leaning towards option 2 and give developers the choice.

**Comment by yjbanov** [comment]
I agree that option 2 is the way to go. If the framework-side API makes assertiveness level an optional parameter, I would also default to assertive. My reasoning is that if developer decides to make a verbal announcement, there may not be a written equivalent of it anywhere in the UI. Otherwise, they would use a live region or some other way to inform the user. If audio announcement is the only source, missing that would cause the user to lose a piece of potentially critical information.
